### PR TITLE
Update tactical-ddd-content.md

### PR DESCRIPTION
**Pull request by azarboon**
Before clicking the "Create pull request" button:

Enter a meaningful title. We suggest using the keywords "New" or "Update" indicating the nature of changes.

#### Summary, scope, and intent of this PR

Added following. [Here](https://youtu.be/Ki9obBrFlrQ) and [here](https://www.ben-morris.com/entity-services-when-microservices-are-worse-than-monoliths/) are sources of my claim.

"Be wary of Entity Trap also known as Entity Service Antipattern wherein services are modeled around entities / data instead of behavior. Entity services either expose simple CRUD methods or end up too big to lose their purpose. Service names such as *Manager, *Supervisor, *Handler, *Helper, *Controller can indicate that the associated service is in an entity trap."

#### Links to any related work item(s) or supporting detail

- [DELETE-OR-REPLACE-THIS-TEXT]
- [DELETE-OR-REPLACE-THIS-TEXT]

  

<details><summary>AFTER YOUR PR HAS BEEN CREATED, expand this section for tips and additional instructions.</summary>
These are common guidelines for contributions across the repos managed by the Azure patterns & practices (PnP) content team. Some repositories may have additional specific requirements that are not listed here.   

## Guidance for all contributors  
  
| Topic | Guidance |
| ----------| ------------ |
| **Draft PR** | If your PR will be a work-in-progress for more than a day or two, select the **Convert to draft** link in the upper right of the page (under **Reviewers**) to change it to a draft. For future reference, you can also do this using the **Create pull request** button drop-down during PR creation. | 
| **ms.date metadata** | <ul><li>Don't update an article's "ms.date" metadata property unless you've done a **full freshness review** of the content. A full freshness review includes changes required to correct or improve the **full** technical accuracy of the article.</li><li>Don't update "ms.date" if you're doing targeted changes to improve non-technical aspects of the article, such as editorial quality, art improvements, article template alignment, etc.</li><li>If you've changed any "ms.date" properties for work that wasn't part of full review for freshness, please reset them to their previous value.</li></ul> | 
| **Placement and linking** | If you're creating a new article or articles, include updates to the related TOC.yml file to propose where the article(s) should be placed. Also consider other places within the document set where it would be beneficial to cross-reference and link to your new article(s). | 
| **PR build** | After you open your PR, and for each successive commit that you push to your branch, the publishing platform will run validation on the files in your pull request. A summary of the build results for each file will be inserted inline into your pull request, which includes any build suggestions/warnings/errors. PRs cannot be merged until all build errors and most warnings are resolved. |
| **Publishing** | Following a successful merge, most repos publish to the live site about once per (business) day. |
| **Additional resources** | <ul><li>[learn.microsoft.com contributor guide](https://review.learn.microsoft.com/help/contribute/?branch=main)</li></ul>
  
## Additional guidance for private repos and internal contributors  

| Topic | Guidance | 
| ----------| ------------ | 
| **PR size** | If your PR is more than ~5 lines of changes, or you'd like for the changes to go through editorial or larger review, open a contribution request at <https://aka.ms/contribution> and include a link to the PR in response #8. Once it's processed, you'll be notified of the next steps.  |
| **PR title prefix** | Select the **Edit** button to the right of the PR title if you need to revise it. The following prefixes are reserved for specific contribution types:<br/><br/><ul><li>**[Quality Check]** - maintenance work related to content quality (edit passes, art improvements, template alignment)</li><li>**[LinkFix]** - recurring/adhoc PRs to correct link URLs</li><li>**[Pipeline]** - new/updated contributor success pipeline content</li><li>**[WIP]** - a work-in-progress draft requiring several days/weeks</li></ul> |
| **PR preview** | Following successful build of your PR, publishable files will also include **Preview URL** links to staged previews of your new/updated articles. Be sure to review these for verification of your intended contributions, or to send to other internal contributors for review. |
| **PR sign-off (public repo)** | If an article you own is updated in a public repo PR, you are responsible for sign-off. You will be automatically notified via email. The PR will not be merged until you've had a chance to review and sign-off. |
| **PR sign-off (private repo)** | After you've completed your proposed changes, addressed build warnings, and completed all review work, you can begin the sign-off process for review and merge:<br/><br/><ol><li>If your PR is in draft mode, remove "[WIP]" from the title and select **Ready for review** button at the bottom of the PR.</li><li>Enter "#sign-off" in a new comment. This comment indicates that **you're confident the work meets or exceeds Microsoft's standards for publication**, and will trigger the review process.</li><li>Your PR may be selected for initial review by the PnP content team. Following PnP review, you may receive questions or requests for additional changes. You should have initial feedback from PnP review within a few business days. If you have an urgent request or need to contact the team, please mention `@MicrosoftDocs/patterns-and-practices-team-pr-reviewers` in your PR and someone will get back to you. After the Azure patterns & practices content review is complete, a `PnP #sign-off` will be added.</li><li>Final review/merge is done by the PR review team. The PR team may also respond with feedback, categorized as "Blocking" (requires action from you), or "Non-blocking" (to be addressed in a future PR).</li></ol> |
| **Additional resources** | <ul><li>[learn.microsoft.com internal contributor guide](https://review.learn.microsoft.com/help/contribute/?branch=main)</li><li>Authoring templates: [architecture-center-pr](https://review.learn.microsoft.com/help/contribute/architecture-center/templates/sample-solution-templates?branch=main), [well-architected-pr](https://review.learn.microsoft.com/help/contribute/global-waf-template?branch=main)</li><li>To contact the Azure patterns & practices team use [e-mail](mailto:cact-pr-reviewers@microsoft.com?subject=Help%20with%20pull%20request), or @mention our GitHub team in your PR comments using: `@MicrosoftDocs/patterns-and-practices-team-pr-reviewers`</li></ul>
</details>

### CODE REVIEW

File: docs/microservices/model/tactical-ddd-content.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ This section provides a brief summary of the tactical DDD patterns, so if you ar
 - The attributes of an entity may change over time. For example, a person's name or address might change, but they are still the same person.
 - An entity can hold references to other entities.
 
+> [!NOTE]
+> Be wary of Entity Trap also known as Entity Service Antipattern wherein services are modeled around entities / data instead of behavior. Entity services either expose simple CRUD methods or end up too big to lose their purpose. Service names such as *Manager, *Supervisor, *Handler, *Helper, *Controller can indicate that the associated service is in an entity trap.
```
Comment (ckittel) [inline docs/microservices/model/tactical-ddd-content.md:19]: You say "don't model around data" but instead make sure you're modeling around "behavior" -- but then all of your service names are behavior.  See what you can do to reword this to bring more clarity.  Do it WITHOUT using the terms "Entity Trap" and "Entity Service Antipattern" - I feel like those might be attempting to be a shortcut to explaining this.

Just explain the problem using non-jargon.

Also, be sure you include what **harm** would come to your application if you modeled parts of your system this way. Meaning, list the answer to "what's the harm?" -- "too simple or too big" are ambiguous harms and I don't know how that would matter by reading this.

**Review by ckittel — CHANGES_REQUESTED** [review]

File: docs/microservices/model/tactical-ddd-content.md (hunk 1 above)
Comment (azarboon) [reply docs/microservices/model/tactical-ddd-content.md]: Thanks for the comment. Changed it. As per Mark Richards, these are the issues: non-describing names, dumping ground for unrelated functionalities, and overloaded services.

However, I just noticed that in the next page, services have been modelled around entities such as accounts, drone management, etc. It seems to me that author of this article have already fallen for entity trap. What's your advice?

Comment (ckittel) [reply docs/microservices/model/tactical-ddd-content.md]: I think your rewrite is better, but the inconsistency that is brought forth in this article by it is troubling.  We don't have bandwidth right now to evaluate this situation.  I'm going to close the PR, so we don't have competing guidance in the same article.  On the page, you can provide feedback.  Click the feedback button at the bottom, give the article as a whole a rating and add this observation to the text.  Drop my name in the text as well, that'll be a reminder that we talked.  We review that feedback and backlog items once a week or so.

**Comment by ckittel** [comment]
#please-close

File: docs/microservices/model/tactical-ddd-content.md (hunk 1 above)
Comment (azarboon) [reply docs/microservices/model/tactical-ddd-content.md]: Sure. I will do as you suggested. Thanks
